src/sparseml/transformers/utils/initializers.py

```diff
@@ -123,12 +120,10 @@ def initialize_sparse_model(
     :param config: the config to use for the model
     :param trust_remote_code: True to trust remote code when loading the model,
         False otherwise
-    :param recipe: the recipe to apply to the model. If None, will look for a recipe
-        in the model_path
+    :param recipe: the recipe to apply to the model.
```

Comment:
Is this change because we are calling this function after we've resolved the recipe? (ie recipe will already point to the recipe in the model_path?)

Reply:
We are calling this function before we've resolved the recipe. The resolve_function will be called later on:
initialize_sparse_model --> load_task_model --> SparseAutoModel.from_pretrained -> resolve_recipe